sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogVersionsSuite.scala

```diff
@@ -50,14 +52,24 @@ class HiveExternalCatalogVersionsSuite extends SparkSubmitTestUtils {
     super.afterAll()
   }
 
-  private def downloadSpark(version: String): Unit = {
-    import scala.sys.process._
+  private def tryDownloadSpark(version: String, path: String): Unit = {
+    // Try mirrors a few times until one succeeds
+    for (i <- 0 until 3) {
+      val preferredMirror =
+        Seq("wget", "https://www.apache.org/dyn/closer.lua?preferred=true", "-q", "-O", "-").!!.trim
+      val url = s"$preferredMirror/spark/spark-$version/spark-$version-bin-hadoop2.7.tgz"
+      logInfo(s"Downloading Spark $version from $url")
```

Comment:
FYI not sure which is the better way, in SparkR we get a list with `http://www.apache.org/dyn/closer.cgi&as_json=1`

```
{
       "backup": [ "http://www-eu.apache.org/dist/", "http://www-us.apache.org/dist/" ],
         "cca2": "us",
          "ftp": [ "ftp://apache.cs.utah.edu/apache.org/", "ftp://apache.mirrors.tds.net/pub/apache.org/", "ftp://ftp.osuosl.org/pub/apache/", "ftp://mirror.reverse.net/pub/apache/" ],
         "http": [ "http://apache.claz.org/", "http://apache.cs.utah.edu/", "http://apache.mesi.com.ar/", "http://apache.mirrors.hoobly.com/", "http://apache.mirrors.ionfish.org/", "http://apache.mirrors.lucidnetworks.net/", "http://apache.mirrors.pair.com/", "http://apache.mirrors.tds.net/", "http://apache.osuosl.org/", "http://download.nextag.com/apache/", "http://ftp.wayne.edu/apache/", "http://mirror.cc.columbia.edu/pub/software/apache/", "http://mirror.cogentco.com/pub/apache/", "http://mirror.jax.hugeserver.com/apache/", "http://mirror.metrocast.net/apache/", "http://mirror.olnevhost.net/pub/apache/", "http://mirror.reverse.net/pub/apache/", "http://mirror.stjschools.org/public/apache/", "http://mirrors.advancedhosters.com/apache/", "http://mirrors.gigenet.com/apache/", "http://mirrors.koehn.com/apache/", "http://mirrors.ocf.berkeley.edu/apache/", "http://mirrors.sonic.net/apache/", "http://supergsego.com/apache/", "http://www.gtlib.gatech.edu/pub/apache/", "http://www.namesdir.com/mirrors/apache/", "http://www.trieuvan.com/apache/" ],
    "path_info": "",
    "preferred": "http://mirror.cc.columbia.edu/pub/software/apache/"
}
```

Reply:
it is possible to also pass `path` but from testing, it doesn't seem like the path/path_info value is validated in any way.